lib/reporters/tap.js

```diff
@@ -52,11 +53,18 @@ function TAP(runner) {
   runner.on('fail', function(test, err) {
     failures++;
     console.log('not ok %d %s', n, title(test));
-    if (err.message) {
-      console.log(err.message.replace(/^/gm, '  '));
-    }
-    if (err.stack) {
-      console.log(err.stack.replace(/^/gm, '  '));
+    var emitYamlBlock = err.message != null || err.stack != null;
+    if (emitYamlBlock) {
+      console.log('  ---');
```

Comment:
Since TAP14 is standardizing on 2 space indention, using something like [this](https://github.com/mochajs/mocha/blob/d401ee198bd5026bc61641ad26eab31e6746713f/lib/reporters/spec.js#L36-L38) might add to maintainability.